src/math/hough_transform_simd.cpp

```diff
@@ -0,0 +1,277 @@
+#include "hough_transform_simd.h"
+#include "../image_function_helper.h"
+#include "../penguinv/cpu_identification.h"
+
+#ifdef PENGUINV_AVX_SET
+#include <immintrin.h>
+#endif
+
+#ifdef PENGUINV_SSE_SET
+#include <emmintrin.h>
+#endif
+
+#ifdef PENGUINV_NEON_SET
+#include <arm_neon.h>
+#endif
+
+#include <algorithm>
+#include <cmath>
+
+namespace
+{
+    const bool float_layout_check  = std::is_standard_layout<PointBase2D<float>>::value;
+    const bool double_layout_check = std::is_standard_layout<PointBase2D<double>>::value;
+    const size_t avx_double = 4u;
+    const size_t avx_float = 8u;
+    const size_t sse_double = 2u;
+    const size_t sse_float = 4u;
+    const size_t neon_double = 2u;
+    const size_t neon_float = 4u;
+
+    const float minimumAngleStep = 0.001f * static_cast<float>( pvmath::pi ) / 180.0f;
+    const float minimumLineTolerance = 1e-5f;
+    
+    template <typename _Type>
+    void FindDistance( const std::vector< PointBase2D< _Type > > & input, std::vector < _Type > & distance, _Type cosVal, _Type sinVal, size_t offset = 0u )
+    {
+        if ( offset >= input.size() )
+            return;
```

Comment:
Here we better throw `imageException` exception instead so we could catch some logic problem during unit tests run.